carbon-design-system/carbon
# feat(react): add disclosure hook

Closes #8819 

Related to: #6633

New component! It's Disclosure!

This is phase 1 of 5ish. Built out Disclosure with Text. Next up will be Menus!

Sketch file: 
- https://ibm.ent.box.com/folder/138835990231

**Testing/Reviewing:**

- Checkout the `unstable_Disclosure` component in Storybook and confirm the three stories work as expected.
  - Body (default)
  - Header and Body (with and without divider)
  - Body and Footer (with and without divider)
- Run the tests
  - `yarn test --watch packages/react/src/components/Disclosure/__tests__/useDisclosure-test.js`
  -  Confirm that all 5 tests pass.
- Confirm that we didn't break `unstable_Popover` in the process.
